daemon/oci_linux.go

```diff
@@ -98,6 +98,20 @@ func setDevices(s *specs.Spec, c *container.Container) error {
 		for _, d := range hostDevices {
 			devs = append(devs, oci.Device(d))
 		}
+		for _, deviceMapping := range c.HostConfig.Devices {
+			if _, err := os.Stat(deviceMapping.PathInContainer); os.IsNotExist(err) {
+				d, _, err := oci.DevicesFromPath(deviceMapping.PathOnHost, deviceMapping.PathInContainer, "rwm")
+				if err != nil {
+					return err
+				}
+				devs = append(devs, d...)
+			} else {
+				if err == nil {
```

Comment:
If a user specified `--privileged --device xxx:xxx`, then `xxx` will anyway be available, so specifying `--device xxx:xxx` is redundant, and it should not cause an error. One option is to just ignore it, the other is to have a warning, something like this maybe:

```golang
 log.Warnf("Specifying a device %s is redundant for privileged mode", deviceMapping.PathOnHost)
```

Reply:
Was thinking about that; `--device` allows setting `r` (read), `w` (write), and `m` (minor) options; what options are used if the device passed through `--device` overlaps with a device on the host?

Reply:
This is a valid concern which I totally overlooked. I think in case restrictive permissions (anything other than the most permissive `rwm`) are given, the code should return an error as it can't enforce the required permissions (e.g. read-only).

What do you think?

Reply:
In case of a conflict, yes. Perhaps a warning is ok, after all, `--privileged` is expected to remove any restrictions, so perhaps lifting the restrictions that were specified to be ignored is ok.

@wenlxie what would your expectations be?

Reply:
> If a user specified --privileged --device xxx:xxx, then xxx will anyway be available, so specifying --device xxx:xxx is redundant, and it should not cause an error. One option is to just ignore it, the other is to have a warning, something like this maybe:

Thanks for the comments, will change the code.

>Was thinking about that; --device allows setting r (read), w (write), and m (mknod) options; what options are used if the device passed through --device overlaps with a device on the host?

@thaJeztah @kolyshkin 
For this case,  I'd like  to ignore the restrictions and respect the `--privileged ` option.
Will add a warning for this case.

Reply:
@kolyshkin It will be at the daemon logs and  user will not see it.

Reply:
you can still print a warning in the cli by inspecting the dev options before they go to the daemon